zerver/views/registration.py

```diff
@@ -51,6 +51,13 @@
 def check_prereg_key_and_redirect(request: HttpRequest, confirmation_key: str) -> HttpResponse:
     # If the key isn't valid, show the error message on the original URL
     confirmation = Confirmation.objects.filter(confirmation_key=confirmation_key).first()
+    # if the prereg user has status revoked the confirmation link should throw an error page
+    confirmation_status_obj = Confirmation.objects.get(confirmation_key=confirmation_key,
+                                                       type=confirmation.type)
+    prereg_user = confirmation_status_obj.content_object
+    revoked_value = confirmation_settings.STATUS_REVOKED
+    if prereg_user.status == revoked_value:
+        return render(request, "zerver/realm_reactivation_link_error.html", context={})
```

Comment:
We definitely don't want to use `realm_reactivation_link_error.html` here - we want an error page with an appropriate error message - but the HTML from `realm_reactivation_link_error.html` can be reused (just put a different error message in the new html file).